runtime/runtime.go

```diff
@@ -425,6 +428,10 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 		return fmt.Errorf("at least one address must be configured in runtime parameters")
 	}
 
+	if !rt.Params.AddrChanged {
+		rt.logger.Info("OPA running on a public (0.0.0.0) network interface. Unless you intend to expose OPA outside of the host, binding to the localhost interface (--addr localhost:8181) is recommended. See https://www.openpolicyagent.org/docs/latest/security/#interface-binding")
```

Comment:
Since we are logging this at `info`, why not just merge this with the `Initializing server` log?

Reply:
How does this look @ashutosh-narkar ? :+1: 

```go
	publicInterfaceWarning := ""
	if !rt.Params.AddrSetByUser {
		publicInterfaceWarning = "OPA running on a public (0.0.0.0) network interface. Unless you intend to expose OPA outside of the host, binding to the localhost interface (--addr localhost:8181) is recommended. See https://www.openpolicyagent.org/docs/latest/security/#interface-binding"
	}

	if rt.Params.DiagnosticAddrs == nil {
		rt.Params.DiagnosticAddrs = &[]string{}
	}

	rt.logger.WithFields(map[string]interface{}{
		"addrs":            *rt.Params.Addrs,
		"diagnostic-addrs": *rt.Params.DiagnosticAddrs,
		"public-interface-warning": publicInterfaceWarning,
	}).Info("Initializing server.")
```

Reply:
We should update the message `Info("Initializing server.")` with the warning if needed. For ex. 

```
{"addrs":[":8181"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server. OPA running on a public .....","time":"2023-03-24T11:07:26-07:00"}
```

Reply:
Done! :+1: